README.md

```diff
@@ -1,6 +1,126 @@
-# Traitlets Configuration
+# Traitlets
 
-A lightweight pure-Python derivative of Enthought Traits,
-used for configuring Python objects.
+[![Build Status](https://travis-ci.org/ipython/traitlets.svg?branch=master)](https://travis-ci.org/ipython/traitlets)
+[![Documentation Status](https://readthedocs.org/projects/traitlets/badge/?version=latest)](http://traitlets.readthedocs.org/en/latest/?badge=latest)
 
-This package powers the config system of IPython and Jupyter.
+Traitlets are a pure Python library enabling
+
+ - the enforcement of strong typing for attributes of Python objects
+(typed attributes are called "traits"),
+ - notifications on changes of trait attributes,
+ - automatic validation and coercion of trait attributes when attempting a
+change.
+
+The implementation relies on the [descriptor](https://docs.python.org/howto/descriptor.html)
+pattern. This package powers the configuration system of IPython and Jupyter,
+and the declarative API of IPython interactive widgets.
+
+## Installation
+
+For a local installation, make sure you have
+[pip installed](https://pip.readthedocs.org/en/stable/installing/) and run:
+
+```
+pip install traitlets
+```
+
+For a development installation:
+
+* Clone this repository and `cd` into it
+* `pip install --pre -e .`
+
+## Running the tests
+
+* `nosetests traitlets`.
+
+## Usage
+
+Any class with trait attributes must inherit from `HasTraits`.
+For the list of available trait types and ther properties, see the
+[Trait Types](http://traitlets.readthedocs.org/en/latest/trait_types.html)
+section of the documentation.
+
+### Dynamic default values
+
+To calculate a default value dynamically, decorate a method of your class with
+@default({traitname})`. This method will be called on the instance, and should
+return the default value. For example:
+
+```Python
+import getpass
+from traitlets import HasTraits, Unicode, observe
+
+class Identity(HasTraits):
+    username = Unicode()
+
+    @default('username')
```

Comment:
I'm going to be annoying, I know, but reading that I'm wondering if for a future iteration it wouldn't be nice to (also) be able to use `@username.default` as a decorator. It would be closer to how property works. Agreed it's not doable on subclass....